test/plugins/collections.test.js

```diff
@@ -289,4 +290,98 @@ describe('collection plugin test', () => {
             });
         });
     });
+
+    describe('PUT /collections/{id}', () => {
+        const id = testCollection.id;
+        let options;
+        let updatedCollectionMock;
+
+        beforeEach(() => {
+            options = {
+                method: 'PUT',
+                url: `/collections/${id}`,
+                payload: {
+                    name: 'updated name',
+                    description: 'updated description',
+                    pipelineIds: [123, 124]
+                },
+                credentials: {
+                    username,
+                    scope: ['user']
+                }
+            };
+
+            updatedCollectionMock = hoek.clone(collectionMock);
+            updatedCollectionMock.name = 'updated name';
+            updatedCollectionMock.description = 'updated description';
+            updatedCollectionMock.pipelineIds = [123, 124];
+
+            collectionFactoryMock.get.withArgs({ id }).resolves(collectionMock);
+            collectionMock.update.resolves(updatedCollectionMock);
+            collectionMock.toJson.returns(updatedCollection);
+        });
+
+        it('returns 200 and correct collection data', () =>
+            server.inject(options).then((reply) => {
+                assert.deepEqual(reply.result, updatedCollection);
+                assert.calledOnce(collectionMock.update);
+                assert.equal(reply.statusCode, 200);
+            })
+        );
+
+        it('returns 200 and correct collection data when no fields modified', () => {
+            options.payload = {};
+            collectionMock.update.resolves(collectionMock);
+            collectionMock.toJson.returns(testCollection);
+
+            return server.inject(options).then((reply) => {
+                assert.deepEqual(reply.result, testCollection);
+                assert.calledOnce(collectionMock.update);
```

Comment:
You're just checking if `collectionMock.update` was called. Is there no check on whether the model's fields were not modified?